src/main/java/com/app/bookstore/service/impl/ShoppingCartServiceImpl.java

```diff
@@ -62,6 +62,12 @@ public void deleteCartItem(Long cartItemId) {
         cartItemRepository.deleteById(cartItemId);
     }
 
+    @Override
+    public void clear(ShoppingCart cart) {
```

Comment:
I'd recommend you put Transactional here. You can have a lot of items, and it could take a while to delete all of them